MachObfuscator/Options/Options.swift

```diff
@@ -31,6 +31,7 @@ struct Options {
     var sourceFileNamesPrefixes: [String] = []
     var cstringsReplacements: [String: String] = [:]
     var obfuscableFilesFilter = ObfuscableFilesFilter.defaultObfuscableFilesFilter()
+    var analyzeDependencies = true
```

Comment:
I wanted this class to be plain-old-stupid. I even thought about implementing CommandLine Decoder and making this class a struct :) Anyway – I think it should express command line options 1:1. This name adds a level of indirection, and makes namespace larger. Would you mind changing this to `xxWithoutDependencies` or `withoutDependencies`?

Reply:
I think that `analyzeDependencies` (or `deepSearch`, as you proposed) is a better name from the point of view of code using it - as it is plain, not negated: `if analyzeDepencencies { /*do sth complicated*/ }` is easier to understand than `if !withoutDependencies { /*do sth complicated*/ }`

The command line option is indeed named different, but that is because `analyzeDependencies` is a default behaviour and the command line option is disabling it.  Similar case is for `eraseSymtab` -> `--preserve-symtab`. Maybe we could have some convention for naming options that disable particular behaviour, like prefixing them with `no-`, eg. `--no-erase-symtab`, `--no-xx-analyze-dependencies`. They are not so nice and a bit harder to read but they map to obvious variables in `Options`.

What do you think?

Reply:
You are right. We already have this inconsistency for other options e.g. `replace-cstring-*`. And this is a kind of internal problem, so we can refactor this anytime later, if we decide to do so.

I like your suggestion about creating a naming convention. I also love you suggestion about using `no-*` syntax. I started wondering if there are any CLI conventions... and there are :) e.g [GNU](https://www.gnu.org/prep/standards/html_node/Command_002dLine-Interfaces.html) – there is even a [table of standard options used by the GNU programs](https://www.gnu.org/prep/standards/html_node/Option-Table.html#Option-Table). It looks like a good way to go, and it contains a lot of `no-*` options :) Maybe we could add those URLs as a reference on top of the `Options` struct? What do you think? Do you have other idea?

Reply:
I like the `no-*` convention, I have seen it in some tools. I changed the option name and added description.